### ISSUE TITLE
about the input?

### ISSUE THREAD

**Issue by abcdvzz** [choasUp/caffe-hex#1]
Hello, I'm here to ask about the iuput data of Cifar-100 network. How can I produce the LMDB using my own data? 
I just cannot understand. If I have three labels: 0-car, 1-suv, 2-mpv and two images: suv.jpg, mpv.jpg, how can I produce my LMDB. 
My first guess is like this one: suv.jpg-0, mpv.jpg-0, suv.jpg-1, mpv.jpg-2. And the second one: suv.jpg-0,1 and mpv.jpg-0,2
Could you please tell me which one is right or how to produce the LMDB? Thank you very much.

**Comment by choasup** [comment]
First you have a tree, then each image is labeled at one node(label).

  0
 /  \
1   2

01.img 0
02.img 1
03.img 2
04.img 1
...

please refer to example/cifar10_hex/convert_data.cpp

**Comment by abcdvzz** [comment]
But each image has two labels: coarse label and fine label.

**Comment by choasup** [comment]
just use finelabel
once you confirm finelabel you know corse label

choas

**Comment by abcdvzz** [comment]
So...Take Cifar-100 dataset for an example, the input only use 0-100 label. And the tree relationship is defined in g_file? But why does fc layer have 121 num_output?

**Comment by choasup** [comment]
In the training data, some image can be labeled at corse label, also can be labeled at fine label. Just telling the deepest label is ok.

In the test, some image maybe predicted at the corse label, so fc layer have 121 num_ouput.

**Comment by abcdvzz** [comment]
So during the training procedure, one image might not a have fine label? And it can be labeled at coarse label this time? It is the first question.
Secondly, If each image has its corresponding fine label, can I just reduce the num_output of fc layer to the number of fine labels?

**Comment by choasup** [comment]
One image might label at corse label.

If you use hex graph maybe you cannot reduce num_output.

Please refer to the paper.

choas
